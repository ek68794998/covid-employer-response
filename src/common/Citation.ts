import { CitationSource } from "./CitationSource";
import { CitationType } from "./CitationType";

export class Citation {
	public positivity: number = 0;

	public sources?: CitationSource[];

	public summary: string = "";

	public type?: CitationType;
}
