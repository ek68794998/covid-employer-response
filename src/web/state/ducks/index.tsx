import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});