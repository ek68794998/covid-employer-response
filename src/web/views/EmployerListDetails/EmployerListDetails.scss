$body-color: #37474f;

div.employer-container {
	border: 2px solid $body-color;
	border-radius: 8px;
}

button.employer-header {
	background: $body-color;
	border: 0;
	color: #eceff1;
	cursor: pointer;
	display: flex;
	outline: none;
	text-align: left;
	width: 100%;

	.employer-title {
		flex: 1;
		font: 400 20px 'Baloo Chettan 2';
	}

	span {
		font-size: 24px;
		opacity: 0.5;
	}
}

ul.employer-details {
	display: flex;
	list-style: none;
	margin: 12px;
	padding: 0;
	width: 100%;

	li {
		align-items: center;
		color: #757575;
		display: flex;
		flex: 1;
		transition: 0.2s all;

		&:hover {
			color: #212121;
		}

		a {
			color: inherit;
		}
	}
}

ul.employer-citations {
	font-size: 16px;
	margin: 12px;
	padding: 0;

	li {
		list-style: none;
		margin-top: 16px;
	}
}
