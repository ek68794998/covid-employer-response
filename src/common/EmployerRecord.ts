import { Citation } from "./Citation";
import { EmployerLocation } from "./EmployerLocation";
import { EmployerRating } from "./EmployerRating";

export class EmployerRecord {
	public citations: Citation[] = [];

	public employeesBeforeMax: number = -1;

	public employeesBeforeMin: number = -1;

	public id: string = "";

	public image?: string;

	public location: EmployerLocation = new EmployerLocation();

	public name: string = "";

	public officialWebsite?: string;

	public rating: EmployerRating = "fair";

	public summary: string = "";

	public wiki?: string;
}
