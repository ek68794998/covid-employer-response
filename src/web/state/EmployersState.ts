import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}