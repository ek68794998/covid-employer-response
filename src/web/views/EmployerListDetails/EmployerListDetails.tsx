import React from "react";
import { useSelector } from "react-redux";
import { RouteProps } from "react-router-dom";

import { EmployerRating } from "../../../common/EmployerRating";
import { EmployerRecord } from "../../../common/EmployerRecord";
import { LocalizedStrings } from "../../../common/LocalizedStrings";

import { getStrings } from "../../state/ducks/localization/selectors";

import EmployerDetailsHeader from "../EmployerDetailsHeader/EmployerDetailsHeader";

import "./EmployerListDetails.scss";

interface Props extends RouteProps {
	employer: EmployerRecord;

	onClick: () => void;
}

const materialIcon = (name: string): JSX.Element => <i className="material-icons">{name}</i>;

const EmployerListDetails: React.FC<Props> = (props: Props): React.ReactElement => {
	const strings: LocalizedStrings = useSelector(getStrings);
	const { employer, onClick } = props;

	const rating: EmployerRating = EmployerRecord.getRating(employer);

	return (
		<div className={`EmployerListDetails__Container EmployerListDetails__Rating--${rating}`}>
			<EmployerDetailsHeader employer={employer} onClickEmployerName={onClick} useShortText={true} />
			<div className="EmployerListDetails__Summary">
				{employer.summary}
				<div className="EmployerListDetails__OverflowScreen" />
			</div>
			<div className="EmployerListDetails__Actions">
				<a href="#" onClick={onClick}>
					{strings.readMore}
					{materialIcon("fullscreen")}
				</a>
			</div>
		</div>
	);
};

export default EmployerListDetails;
