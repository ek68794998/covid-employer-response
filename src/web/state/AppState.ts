import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}