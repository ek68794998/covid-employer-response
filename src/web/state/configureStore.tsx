import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;