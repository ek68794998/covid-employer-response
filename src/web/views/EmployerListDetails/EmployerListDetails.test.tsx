import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { mockComponent } from "../../../__tests__/TestUtils";
import { EmployerRecord } from "../../../common/EmployerRecord";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import EmployerListDetails from "./EmployerListDetails";

jest.mock(
	"../EmployerDetailsHeader/EmployerDetailsHeader",
	() => mockComponent("EmployerDetailsHeader"));

describe("<EmployerListDetails />", () => {
	test("renders without exploding", () => {
		const store: Store<AppState, AnyAction> = configureStore({
			strings: {
				readMore: "řèáδMôřè",
			},
		});

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<EmployerListDetails employer={new EmployerRecord()} onClick={(): void => { /* Do nothing. */ }} />
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
