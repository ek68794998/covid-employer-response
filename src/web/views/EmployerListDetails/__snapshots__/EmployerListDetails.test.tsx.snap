// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployerListDetails /> renders without exploding 1`] = `
<div
  className="EmployerListDetails__Container EmployerListDetails__Rating--fair"
>
  [component: EmployerDetailsHeader employer=([object Object]) onClickEmployerName=(() → { }) useShortText=(true) /]
  <div
    className="EmployerListDetails__Summary"
    onClick={[Function]}
  >
    
    <div
      className="EmployerListDetails__OverflowScreen"
    />
  </div>
  <div
    className="EmployerListDetails__Actions"
  >
    <a
      className="EmployerActionLinks__Link"
      href="https://github.com/ekumlin/covid-employer-response/blob/master/public/employers/.yml"
      target="_blank"
      title="edit string"
    >
      <i
        className="material-icons"
      >
        edit
      </i>
    </a>
    <a
      className="EmployerListDetails__ReadMore"
      onClick={[Function]}
    >
      řèáδMôřè
      <i
        className="material-icons"
      >
        fullscreen
      </i>
    </a>
  </div>
</div>
`;
